pkg/asset/installconfig/powervs/session.go

```diff
@@ -19,105 +24,201 @@ var (
+// A bluemix session object may be a better fit here
 type Session struct {
 	Session *ibmpisession.IBMPISession
 }
 
-// GetSession returns an IBM Cloud session by using credentials found in default locations in order:
-// env IBMID & env IBMID_PASSWORD,
-// ~/.bluemix/config.json ? (see TODO below)
-// and, if no creds are found, asks for them
-/* @TODO: if you do an `ibmcloud login` (or in my case ibmcloud login --sso), you get
-//  a very nice creds file at ~/.bluemix/config.json, with an IAMToken. There's no username,
-//  though (just the account's owner id, but that's not the same). It may be necessary
-//  to use the IAMToken vs the password env var mentioned here:
-//  https://github.com/IBM-Cloud/power-go-client#ibm-cloud-sdk-for-power-cloud
-//  Yes, I think we'll need to use the IAMToken. There's a two-factor auth built into the ibmcloud login,
-//  so the password alone isn't enough. The IAMToken is generated as a result. So either:
-     1) require the user has done this already and pull from the file
-     2) ask the user to paste in their IAMToken.
-     3) let the password env var be the IAMToken? (Going with this atm since it's how I started)
-     4) put it into Platform {userid: , iamtoken: , ...}
-*/
+// PISessionVars is an object that holds the variables required to create an ibmpisession object
+type PISessionVars struct {
+	ID     string `json:"id,omitempty"`
+	APIKey string `json:"apikey,omitempty"`
+	Region string `json:"region,omitempty"`
+	Zone   string `json:"zone,omitempty"`
+}
+
+// GetSession returns an ibmpisession object
 func GetSession() (*Session, error) {
 	s, err := getPISession()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to load credentials")
 	}
-
 	return &Session{Session: s}, nil
 }
 
-/*
-//  https://github.com/IBM-Cloud/power-go-client/blob/master/ibmpisession/ibmpowersession.go
-*/
+var (
+	defaultAuthFilePath = filepath.Join(os.Getenv("HOME"), ".powervs", "config.json")
+)
+
 func getPISession() (*ibmpisession.IBMPISession, error) {
 
-	var (
-		id, iamtoken, apikey, region, zone string
-	)
+	var err error
+	var pisv PISessionVars
+
+	// First grab variables from the installer written authFilePath
+	logrus.Debug("Gathering variables from AuthFile")
+	err = getPISessionVarsFromAuthFile(&pisv)
+	if err != nil {
+		return nil, err
+	}
+
+	// Second grab variables from files, currently there are no other files to check
+
+	// Third grab variables from the users enviornment
+	logrus.Debug("Gathering variables from user enviornment")
+	err = getPISessionVarsFromEnv(&pisv)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fourth prompt the user for the remaining variables
+	logrus.Debug("Gathering variables from user")
+	err = getPISessionVarsFromUser(&pisv)
+	if err != nil {
+		return nil, err
+	}
+
+	// Save variables to disk
+	err = savePISessionVars(&pisv)
+	if err != nil {
+		return nil, err
+	}
+
+	logrus.Debug("Done")
```

Comment:
let's remove this or make it more descriptive. you could output a generic Done, but then still return an error below